code/modules/admin/smites/imaginary_friend_special.dm

```diff
@@ -10,34 +14,61 @@
  **/
 /datum/smite/custom_imaginary_friend
 	name = "Imaginary Friend (Special)"
-	var/client/friend_candidate_client
+	/// Who are we going to add to your head today?
+	var/list/friend_candidates
+	/// Do we randomise friend appearances or not?
 	var/random_appearance
 
 /datum/smite/custom_imaginary_friend/configure(client/user)
-	friend_candidate_client = tgui_input_list(user, "Pick the player to put in control", "New Imaginary Friend", sort_list(GLOB.clients))
-	if(isnull(friend_candidate_client))
+	var/appearance_choice = tgui_alert(user,
+		"Do you want the imaginary friend(s) to share name and appearance with their currently selected character preferences?",
+		"Imaginary Friend Appearance?",
+		list(CHOICE_PREFS_APPEARANCE, CHOICE_RANDOM_APPEARANCE, "Cancel"))
+	if (appearance_choice != CHOICE_PREFS_APPEARANCE && appearance_choice != CHOICE_RANDOM_APPEARANCE)
+		return FALSE
+	random_appearance = appearance_choice == CHOICE_RANDOM_APPEARANCE
+
+	var/picked_client = tgui_input_list(user, "Pick the player to put in control", "New Imaginary Friend", list(CHOICE_POLL_GHOSTS) + sort_list(GLOB.clients))
+	if(isnull(picked_client))
 		return FALSE
 
+	if(picked_client == CHOICE_POLL_GHOSTS)
+		var/how_many = tgui_input_number(user, "How many imaginary friends should be added?", "Imaginary friend count", default = 1, min_value = 1)
+		if (isnull(how_many) || how_many < 1)
+			return FALSE
+		var/list/volunteers = poll_ghost_candidates(
+			question = "Do you want to play as an imaginary friend?",
+			jobban_type = ROLE_PAI,
+			poll_time = 10 SECONDS,
+			ignore_category = POLL_IGNORE_IMAGINARYFRIEND,
+		)
+		var/volunteer_count = length(volunteers)
+		if (volunteer_count == 0)
+			to_chat(user, span_warning("No candidates volunteered, aborting."))
+			return FALSE
+		shuffle_inplace(volunteers)
+		friend_candidates = list()
+		while (how_many > 0 && length(volunteers) > 0)
+			var/mob/dead/observer/lucky_ghost = pop(volunteers)
+			if (!lucky_ghost.client)
+				continue
+			how_many--
+			friend_candidates += lucky_ghost.client
```

Comment:
should we split this out into its own proc? seems to be a really long and drawn out interruption in this centralized proc. just do something like `return poll_ghosts()` where `poll_ghosts()` is a subtype-level proc that will return TRUE/FALSE